### DISCUSSION TITLE
PWM function to DigitalOut and back during runtime

### DISCUSSION THREAD

**Discussion by PaluZoli** [arendst/Tasmota#18661] [Q&A]
Hi,
I have a PWM based dimmer realised with Tasmota. It is controlled with WEB commands. The commands are comming all day with 10s repetition rate. But there are Hours were I would like to set 100% or full on regardles of the WEB commands.
Is it possible to disable the WEB command interface with Rule?
Is it possible to define the pin as Digital Output from PWM and back durig run time with a Rule?
Is it possible to periodicly owerwrite the external PWM command. ( At each 100ms write PWM 100%, the one 100ms where it is from WEB interface I can accept.)
Thanks for any suggestions.

**Comment by sfromis** [comment]
If what you mean is that you have some automation doing commands, you could teach that automation to behave. Or you could have it using commands to only invoke a rule, which then could do what's relevant - or not.

It may also be useful to explain what you're trying to do, instead of asking for solutions without revealing what is to be solved.

**Comment by PaluZoli** [comment]
I have a smart power meter which will send every 10s the DTY. With tasmota I will use this PWM/DTY to heat water. I have PV at home and if I have extra current I will use it for heating the water. Since everything is wifi connected I tend to realize some safety. If the smart meter controls the tasmota and a signal fails it could happen that i heat with 100% when there is no extra current available.
Somehow I need to set the tasmota PWM back to 0%, also when I need warm water but it is not warm enough to heat it extra with 100%.
So most of the time it should receive the web commands, but in other cases it should not react on them, but remain 0% or 100%.

**Reply by sfromis** [reply]
What do you mean by "send the DTY" (not a Tasmota term)? How does it get into Tasmota? Also "web commands" is not an unambiguous term. Do you mean `http://tasmotadevice/cm?cmnd=command` or what? If the latter, you can send an Event command to invoke a rule, as I suggested.

**Comment by PaluZoli** [comment]
Tasmota has a PWM command, where you give the duty cycle (DTY) of ON as a percent / number. Im sendig that DTY with web request.
Otherwise thanks for the support. I have solved my issue. Since Im a HW developer, not a SW dev. I have found an easy HW fix.
I use now 2 GPIOs. One is PWM other is POWER[x] output. The two signals are connected together in an OR gate. Which output is the desired signal for me.
With 10s repetition rate there is a PWM through web request. And when I need full operation I can set the POWER[x] to High (ON) and it overrides the PWM.

**Reply by sfromis** [reply]
While you of course can make hardware solutions to things not needing to be done in hardware, it does loose you a lot of flexibility if you want to change the logic later.
